internal/testplanet/uplink.go

```diff
@@ -342,75 +332,94 @@ func atLeastOne(value int) int {
+	libuplinkCfg.Volatile.TLS = struct {
+		SkipPeerCAWhitelist bool
+		PeerCAWhitelistPath string
+	}{
+		SkipPeerCAWhitelist: !config.TLS.UsePeerCAWhitelist,
+		PeerCAWhitelistPath: config.TLS.PeerCAWhitelistPath,
 	}
 
-	// ToDo: Handle Versioning for Uplinks here
+	libuplinkCfg.Volatile.DialTimeout = config.Client.DialTimeout
+	libuplinkCfg.Volatile.RequestTimeout = config.Client.RequestTimeout
 
-	tc := transport.NewClientWithTimeouts(tlsOpts, transport.Timeouts{
-		Request: config.Client.RequestTimeout,
-		Dial:    config.Client.DialTimeout,
-	})
+	return libuplink.NewUplink(ctx, libuplinkCfg)
+}
 
-	if config.Client.SatelliteAddr == "" {
-		return nil, nil, cleanup, errs.New("satellite address not specified")
+// GetProject returns a libuplink.Project which allows interactions with a specific project
+func (planetUplink *Uplink) GetProject(ctx context.Context, satellite *satellite.Peer) (*libuplink.Project, error) {
+	testLibuplink, err := planetUplink.NewLibuplink(ctx)
+	if err != nil {
+		return nil, err
 	}
-
-	m, err := metainfo.Dial(ctx, tc, config.Client.SatelliteAddr, config.Client.APIKey)
+	defer func() { err = errs.Combine(err, testLibuplink.Close()) }()
+	planetUplinkAPIKey := planetUplink.APIKey[satellite.ID()]
+	key, err := libuplink.ParseAPIKey(planetUplinkAPIKey)
 	if err != nil {
-		return nil, nil, cleanup, errs.New("failed to connect to metainfo service: %v", err)
+		return nil, err
 	}
-	defer func() {
-		if err != nil {
-			// close metainfo if any of the setup fails
-			err = errs.Combine(err, m.Close())
-		}
-	}()
 
-	project, err := kvmetainfo.SetupProject(m)
+	project, err := testLibuplink.OpenProject(ctx, satellite.Addr(), key)
 	if err != nil {
-		return nil, nil, cleanup, errs.New("failed to create project: %v", err)
+		return nil, errs.Combine(err, testLibuplink.Close())
 	}
+	return project, nil
+}
 
-	ec := ecclient.NewClient(log.Named("ecclient"), tc, config.RS.MaxBufferMem.Int())
-	fc, err := infectious.NewFEC(config.RS.MinThreshold, config.RS.MaxThreshold)
+// GetProjectAndBucket returns a libuplink.Project and Bucket which allows interactions with a specific project and its buckets
+func (planetUplink *Uplink) GetProjectAndBucket(ctx context.Context, satellite *satellite.Peer, bucketName string, planetUplinkCfg uplink.Config) (*libuplink.Project, *libuplink.Bucket, error) {
+	project, err := planetUplink.GetProject(ctx, satellite)
 	if err != nil {
-		return nil, nil, cleanup, errs.New("failed to create erasure coding client: %v", err)
+		return nil, nil, err
 	}
-	rs, err := eestream.NewRedundancyStrategy(eestream.NewRSScheme(fc, config.RS.ErasureShareSize.Int()), config.RS.RepairThreshold, config.RS.SuccessThreshold)
+
+	access, err := setup.LoadEncryptionAccess(ctx, planetUplinkCfg.Enc)
 	if err != nil {
-		return nil, nil, cleanup, errs.New("failed to create redundancy strategy: %v", err)
+		return nil, nil, errs.Combine(err, project.Close())
```

Comment:
maybe use defer to close `project` instead of manual closing with each error?

Reply:
i only want to close if there is an error, otherwise don't close.

Reply:
add named `error` result and then you can do just this:
```
	defer func() {
		if err != nil {
			err = errs.Combine(err, project.Close())
		}
	}()
```